### PR TITLE
fix: displaying user idle

### PR DESCRIPTION
**Pull request by dawidgil**
fix for: https://energyweb.atlassian.net/browse/SWTCH-1415
I updated library responsible for user idle functionality. Also I updated angular to latest 13 version + some other libraries. Had compatibility issues (not all are adjusted, but it's better than it was 😅 )

### CODE REVIEW

**Comment by jrhender** [comment]
@dawidgil can you rebase this PR with `develop` when you have a chance please?

**Comment by dawidgil** [comment]
> @dawidgil can you rebase this PR with `develop` when you have a chance please?

@jrhender rebased.

File: angular.json [hunk 1]
```diff
@@ -170,12 +170,32 @@
...
+                  "with": "src/environments/environment.dev.ts"
+                }
+              ],
+              "optimization": true,
+              "outputHashing": "all",
+              "sourceMap": false,
+              "namedChunks": false,
+              "extractLicenses": true,
+              "vendorChunk": false,
+              "buildOptimizer": true,
```
Comment (jrhender) [inline angular.json:185]: Are these changes necessary because of the updates to the dependency versions?

File: src/app/modules/did-book/components/add-single-record/add-single-record.component.spec.ts [hunk 2]
```diff
@@ -11,19 +11,17 @@ describe('AddSingleRecordComponent', () => {
...
+    TestBed.configureTestingModule({
+      declarations: [AddSingleRecordComponent],
+      providers: [
+        { provide: DidBookService, useValue: didBookServiceSpy },
+        { provide: MatDialogRef, useValue: dialogSpy },
+        { provide: MAT_DIALOG_DATA, useValue: { did: 'did' } },
+      ],
+      schemas: [NO_ERRORS_SCHEMA],
+    }).compileComponents();
+  }));
```
Comment (jrhender) [inline src/app/modules/did-book/components/add-single-record/add-single-record.component.spec.ts:24]: It seems like most of the files are changed because of a code style change related to new lines for brackets. Is this style change relevant to the "user idle" change?

Comment (dawidgil) [reply src/app/modules/did-book/components/add-single-record/add-single-record.component.spec.ts:24]: Yeah, i saw it also. Prettier starts failing for now reason. I did not update anything connected to it. It was also failing on GHA.

File: angular.json (hunk 1 above)
Comment (dawidgil) [reply angular.json:185]: The issue was that not always user idle was showing up. I created a "prod build" for develop to test it. I can revert it, to still have easier to debug, but not optimised version on dev env.

File: src/app/modules/did-book/components/add-single-record/add-single-record.component.spec.ts (hunk 2 above)
Comment (dawidgil) [reply src/app/modules/did-book/components/add-single-record/add-single-record.component.spec.ts:24]: https://github.com/energywebfoundation/switchboard-dapp/runs/7914323996?check_suite_focus=true this one was failing. After changing all this, it is passing. Without this changes in spec files it was also failing on my local version.

Comment (jrhender) [reply src/app/modules/did-book/components/add-single-record/add-single-record.component.spec.ts:24]: Ah okay, I see

File: src/app/layout/header/header.component.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export class HeaderComponent implements OnInit, OnDestroy {
   ) {}
 
   async ngOnDestroy(): Promise<void> {
-    this._subscription$.next();
+    this._subscription$.next(null);
```
Comment (jrhender) [inline src/app/layout/header/header.component.ts:62]: Why this change?

File: src/app/layout/layout.module.ts [hunk 4]
```diff
@@ -30,8 +30,8 @@ import { RawDataModule } from '../modules/raw-data/raw-data.module';
     NgxSpinnerModule,
     ClipboardModule,
     UserIdleModule.forRoot({
-      idle: environment.userIdle,
-      timeout: environment.userIdle,
+      idle: 60,
+      timeout: 30,
```
Comment (jrhender) [inline src/app/layout/layout.module.ts:34]: Was the intention to hardcode these? My first thought is that using the environment is better

File: src/environments/constants.ts [hunk 5]
```diff
@@ -1,4 +1,3 @@
 export const constants = {
   trxRetry: 60, // 1 minute
-  userIdle: 900, // 15 mins in seconds
```
Comment (jrhender) [inline src/environments/constants.ts:3]: Was this value not used previously?

File: src/app/layout/header/header.component.ts (hunk 3 above)
Comment (dawidgil) [reply src/app/layout/header/header.component.ts:62]: Yeah, to update some libs i had to update rxjs. There was a breaking change, that Subject needs to emit value.

File: src/environments/constants.ts (hunk 5 above)
Comment (dawidgil) [reply src/environments/constants.ts:3]: Updated. Forgot to revert it back after testing. Thanks for catching this! 👍

File: src/environments/constants.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
...
+  idleTime: 900,
+  idleTimeout: 300,
```
Comment (jrhender) [inline src/environments/constants.ts:4]: What is the different between `idleTime` and `idleTimeout`? I was looking as the document https://github.com/rednez/angular-user-idle but they don't really explain it

**Review by jrhender — APPROVED** [review]
Looks good to me

File: src/environments/constants.ts (hunk 6 above)
Comment (dawidgil) [reply src/environments/constants.ts:4]: I was looking on this: https://github.com/rednez/angular-user-idle#service-logic
So idle time is how long user can be inactive before starting timeout.
